From aceaf03e6a02b55296419ec807157dc7c2e6a559 Mon Sep 17 00:00:00 2001
From: Gustavo Duarte <gduarte@redhat.com>
Date: Mon, 13 Oct 2014 22:29:57 -0400
Subject: [misc] genwqe: Accidently casting to u32 where u64 is required

Message-id: <1413239416-7778-10-git-send-email-gduarte@redhat.com>
Patchwork-id: 97540
O-Subject: [RHEL7.1 PATCH BZ 1088491 09/28 v2] GenWQE: Accidently casting to u32 where u64 is required
Bugzilla: 1088491
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088491
Upstream Status: a45a0258d1f817b60fcd5bc67dc7ef692f4615bb

commit a45a0258d1f817b60fcd5bc67dc7ef692f4615bb
Author: Frank Haverkamp <haver@linux.vnet.ibm.com>
Date:   Fri Dec 20 20:27:20 2013 +0100

    GenWQE: Accidently casting to u32 where u64 is required

    Fix a casting to u32 where u64 would be appropriate.
    The bad casting made the driver unusable.

    Signed-off-by: Frank Haverkamp <haver@linux.vnet.ibm.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/genwqe/card_utils.c b/drivers/misc/genwqe/card_utils.c
index 663157b..6b1a6ef 100644
--- a/drivers/misc/genwqe/card_utils.c
+++ b/drivers/misc/genwqe/card_utils.c
@@ -59,7 +59,7 @@ int __genwqe_writeq(struct genwqe_dev *cd, u64 byte_offs, u64 val)
  if (cd->mmio == NULL)
   return -EIO;
 
- __raw_writeq((__force u32)cpu_to_be64(val), cd->mmio + byte_offs);
+ __raw_writeq((__force u64)cpu_to_be64(val), cd->mmio + byte_offs);
  return 0;
 }
 
-- 
1.7.1